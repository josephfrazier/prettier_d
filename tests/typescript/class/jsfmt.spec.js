run_spec(__dirname, ["typescript"], {
  disableBabelTS: ["constructor.ts", "generics.ts", "methods.ts"],
});
