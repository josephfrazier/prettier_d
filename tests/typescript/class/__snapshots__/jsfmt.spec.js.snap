// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`constructor.ts format 1`] = `
====================================options=====================================
parsers: ["typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
class foo {
  constructor(static a: number) {}
}

class foo {
  constructor(export a: number) {}
}

class A {
    'constructor': typeof A
    static Foo() {
        return new A()
    }
}

=====================================output=====================================
class foo {
  constructor(static a: number) {}
}

class foo {
  constructor(export a: number) {}
}

class A {
  "constructor": typeof A;
  static Foo() {
    return new A();
  }
}

================================================================================
`;

exports[`dunder.ts format 1`] = `
====================================options=====================================
parsers: ["typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
class F<__T> {}

=====================================output=====================================
class F<__T> {}

================================================================================
`;

exports[`extends_implements.ts format 1`] = `
====================================options=====================================
parsers: ["typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
class Class extends AbstractClass implements Interface1, Interface2, Interface3, Interface4 {}

class ExtendsAbstractClassAndImplementsInterfaces1 extends AbstractClass
  implements Interface1, Interface2, Interface3, Interface4 {}

class ExtendsAbstractClassAndImplementsInterfaces2
  extends AAAAAAAAAAAAAAbstractClass
  implements Interface1, Interface2, Interface3, Interface4 {}

class ExtendsAbstractClassAndImplementsInterfaces3
  extends AAAAAAAAAAAAAAbstractClass
  implements
    Interface1,
    Interface2,
    Interface3,
    Interface4,
    Interface5,
    Interface6,
    Interface7,
    Interface8 {}

class ExtendsAbstractClassAndImplementsInterfaces4
  extends AAAAAAAAAAAAAAbstractClass<Type1, Type2, Type3, Type4, Type5, Type6, Type7> {}

class ExtendsAbstractClassAndImplementsInterfaces5
  extends AAAAAAAAAAAAAAbstractClass<Type1, Type2, Type3, Type4, Type5, Type6, Type7>
  implements
    Interface1,
    Interface2,
    Interface3,
    Interface4,
    Interface5,
    Interface6,
    Interface7,
    Interface8 {}

class ImplementsInterfaceAndExtendsAbstractClass1<Foo>
  extends FOOOOOOOOOOOOOOOOO
  implements FOOOOOOOOOOOOOOOOO, BARRRRRRRRRR {}

class Foo<FOOOOOOOOOOOOOOOOOOOOOOOOOOO, FOOOOOOOOOOOOOOOOOOOOOOOOOOO> implements Foo {}

class ImplementsInterfaceAndExtendsAbstractClass2<
    TypeArgumentNumberOne,
    TypeArgumentNumberTwo,
    TypeArgumentNumberThree
  >
   extends FOOOOOOOOOOOOOOOOOO implements BaseInterface {}

class ImplementsInterfaceClass1<
    TypeArgumentNumberOne,
    TypeArgumentNumberTwo,
    TypeArgumentNumberThree
  >
    implements BaseInterface {}

class ImplementsInterfaceClassWithComments1<
    TypeArgumentNumberOne,
    TypeArgumentNumberTwo,
    TypeArgumentNumberThree
  > // comments
    implements BaseInterface {}

=====================================output=====================================
class Class
  extends AbstractClass
  implements Interface1, Interface2, Interface3, Interface4 {}

class ExtendsAbstractClassAndImplementsInterfaces1
  extends AbstractClass
  implements Interface1, Interface2, Interface3, Interface4 {}

class ExtendsAbstractClassAndImplementsInterfaces2
  extends AAAAAAAAAAAAAAbstractClass
  implements Interface1, Interface2, Interface3, Interface4 {}

class ExtendsAbstractClassAndImplementsInterfaces3
  extends AAAAAAAAAAAAAAbstractClass
  implements
    Interface1,
    Interface2,
    Interface3,
    Interface4,
    Interface5,
    Interface6,
    Interface7,
    Interface8 {}

class ExtendsAbstractClassAndImplementsInterfaces4 extends AAAAAAAAAAAAAAbstractClass<
  Type1,
  Type2,
  Type3,
  Type4,
  Type5,
  Type6,
  Type7
> {}

class ExtendsAbstractClassAndImplementsInterfaces5
  extends AAAAAAAAAAAAAAbstractClass<
    Type1,
    Type2,
    Type3,
    Type4,
    Type5,
    Type6,
    Type7
  >
  implements
    Interface1,
    Interface2,
    Interface3,
    Interface4,
    Interface5,
    Interface6,
    Interface7,
    Interface8 {}

class ImplementsInterfaceAndExtendsAbstractClass1<Foo>
  extends FOOOOOOOOOOOOOOOOO
  implements FOOOOOOOOOOOOOOOOO, BARRRRRRRRRR {}

class Foo<FOOOOOOOOOOOOOOOOOOOOOOOOOOO, FOOOOOOOOOOOOOOOOOOOOOOOOOOO>
  implements Foo {}

class ImplementsInterfaceAndExtendsAbstractClass2<
    TypeArgumentNumberOne,
    TypeArgumentNumberTwo,
    TypeArgumentNumberThree
  >
  extends FOOOOOOOOOOOOOOOOOO
  implements BaseInterface {}

class ImplementsInterfaceClass1<
  TypeArgumentNumberOne,
  TypeArgumentNumberTwo,
  TypeArgumentNumberThree
> implements BaseInterface {}

class ImplementsInterfaceClassWithComments1<
    TypeArgumentNumberOne,
    TypeArgumentNumberTwo,
    TypeArgumentNumberThree
  > // comments
  implements BaseInterface {}

================================================================================
`;

exports[`generics.ts format 1`] = `
====================================options=====================================
parsers: ["typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
class<T> implements Map<T> {}

interface AudioBufferList {
	mBuffers: interop.Reference<any /*AudioBuffer*/>;
}

=====================================output=====================================
class<T> implements Map<T> {}

interface AudioBufferList {
  mBuffers: interop.Reference<any /*AudioBuffer*/>;
}

================================================================================
`;

exports[`methods.ts format 1`] = `
====================================options=====================================
parsers: ["typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
class X {
    optionalMethod?() {}
}

interface Iterable<T> {
  export [Symbol.iterator](): Iterator<T>;
}

export class Check {
  private static property = 'test';
}

=====================================output=====================================
class X {
  optionalMethod?() {}
}

interface Iterable<T> {
  export [Symbol.iterator](): Iterator<T>;
}

export class Check {
  private static property = "test";
}

================================================================================
`;

exports[`optional.ts format 1`] = `
====================================options=====================================
parsers: ["typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
class X {
  private foo? = undefined;
  "a-prop"?: boolean;
}

class A {
  protected [s]?() {}
}

=====================================output=====================================
class X {
  private foo? = undefined;
  "a-prop"?: boolean;
}

class A {
  protected [s]?() {}
}

================================================================================
`;

exports[`quoted-property.ts format 1`] = `
====================================options=====================================
parsers: ["typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
class User {
    "username": string;
}

=====================================output=====================================
class User {
  "username": string;
}

================================================================================
`;

exports[`standard_private_fields.ts format 1`] = `
====================================options=====================================
parsers: ["typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
class Square {
    #sideLength: number;
    readonly #area: number;
    #unit?: string;

    constructor(sideLength: number, unit?: string) {
        this.#sideLength = sideLength;
        this.#area = this.#sideLength ** 2;
        if (unit) {
          this.#unit = unit;
        }
    }

    equals(other: any) {
        return this.#sideLength === other.#sideLength;
    }

    getArea() {
      return this.#area + (this.#unit ?? 'px') + '²';
    }
}

=====================================output=====================================
class Square {
  #sideLength: number;
  readonly #area: number;
  #unit?: string;

  constructor(sideLength: number, unit?: string) {
    this.#sideLength = sideLength;
    this.#area = this.#sideLength ** 2;
    if (unit) {
      this.#unit = unit;
    }
  }

  equals(other: any) {
    return this.#sideLength === other.#sideLength;
  }

  getArea() {
    return this.#area + (this.#unit ?? "px") + "²";
  }
}

================================================================================
`;
